### PR TITLE
Implemented Ad-Hoc observation results API and store functions

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2714** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2714?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2714?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/main/kotlin/com/terraformation/backend/tracking/api/ObservationsController.kt [hunk 1]
```diff
@@ -1014,6 +1034,11 @@ data class CompleteAdHocObservationRequestPayload(
     val swCorner: Point,
 )
 
+data class CompleteAdHocObservationResponsePayload(
```
Comment (tommylau523) [inline src/main/kotlin/com/terraformation/backend/tracking/api/ObservationsController.kt:1042]: Re-ordered these alphabetically. Can be safely ignored for most of these.

File: src/main/kotlin/com/terraformation/backend/tracking/api/ObservationsController.kt [hunk 2]
```diff
@@ -1022,10 +1047,46 @@ data class CompletePlotObservationRequestPayload(
...
+
 data class ListAssignedPlotsResponsePayload(
     val plots: List<AssignedPlotPayload>,
 ) : SuccessResponsePayload
 
+data class ListObservationResultsResponsePayload(
+    val observations: List<ObservationResultsPayload>
+) : SuccessResponsePayload
+
+data class ListObservationSummariesResponsePayload(
```
Comment (tommylau523) [inline src/main/kotlin/com/terraformation/backend/tracking/api/ObservationsController.kt:1074]: This one is renamed to fit with the existing schemas

File: src/main/kotlin/com/terraformation/backend/tracking/api/ObservationsController.kt [hunk 3]
```diff
@@ -1022,10 +1047,46 @@ data class CompletePlotObservationRequestPayload(
...
 )
 
+data class GetObservationResultsResponsePayload(val observation: ObservationResultsPayload) :
+    SuccessResponsePayload
+
+data class ListAdHocObservationsResponsePayload(
+    val observations: List<ObservationPayload>,
+) : SuccessResponsePayload
+
+data class ListAdHocObservationResultsResponsePayload(
```
Comment (tommylau523) [inline src/main/kotlin/com/terraformation/backend/tracking/api/ObservationsController.kt:1062]: This is a new data class

**Review by sgrimm — APPROVED** [review]

**Comment by sgrimm** [comment]
Nit: "functions" is misspelled in the PR title.

**Pull request merged**